test/extensions/http/header_validators/envoy_default/header_validator_test.cc

```diff
@@ -16,80 +9,281 @@ namespace HeaderValidators {
+}
+
+TEST_F(BaseHeaderValidatorTest, ValidateResponseStatusRangeInvalidMin) {
+  auto mode = BaseHttpHeaderValidator::StatusPseudoHeaderValidationMode::ValueRange;
+  HeaderString invalid_min{"99"};
+  auto uhv = createBase(empty_config);
+
+  EXPECT_REJECT_WITH_DETAILS(uhv->validateStatusHeader(mode, invalid_min),
+                             UhvResponseCodeDetail::get().InvalidStatus);
+}
+
+TEST_F(BaseHeaderValidatorTest, ValidateResponseStatusRangeInvalidMax) {
+  auto mode = BaseHttpHeaderValidator::StatusPseudoHeaderValidationMode::ValueRange;
+  HeaderString invalid_max{"1024"};
+  auto uhv = createBase(empty_config);
+
+  EXPECT_REJECT_WITH_DETAILS(uhv->validateStatusHeader(mode, invalid_max),
+                             UhvResponseCodeDetail::get().InvalidStatus);
+}
+
+TEST_F(BaseHeaderValidatorTest, ValidateResponseStatusOfficalCodes) {
+  auto mode = BaseHttpHeaderValidator::StatusPseudoHeaderValidationMode::OfficialStatusCodes;
+  HeaderString valid{"200"};
+  HeaderString invalid_unregistered{"420"};
+  auto uhv = createBase(empty_config);
+
+  EXPECT_TRUE(uhv->validateStatusHeader(mode, valid).ok());
+  EXPECT_REJECT_WITH_DETAILS(uhv->validateStatusHeader(mode, invalid_unregistered),
+                             UhvResponseCodeDetail::get().InvalidStatus);
+}
+
+TEST_F(BaseHeaderValidatorTest, ValidateGenericHeaderNameValid) {
+  HeaderString valid{"x-foo"};
+  auto uhv = createBase(reject_headers_with_underscores_config);
+
+  EXPECT_TRUE(uhv->validateGenericHeaderName(valid).ok());
+}
+
+TEST_F(BaseHeaderValidatorTest, ValidateGenericHeaderKeyRejectUnderscores) {
+  HeaderString invalid_underscore{"x_foo"};
+  auto uhv = createBase(reject_headers_with_underscores_config);
+
+  EXPECT_REJECT_WITH_DETAILS(uhv->validateGenericHeaderName(invalid_underscore),
+                             UhvResponseCodeDetail::get().InvalidUnderscore);
 }
 
-TEST_F(HeaderValidatorTest, Http1RequestHeaderMapValidation) {
-  auto uhv = create(empty_config, ::Envoy::Http::Protocol::Http11);
-  ::Envoy::Http::TestRequestHeaderMapImpl request_header_map{
-      {":method", "GET"}, {":path", "/"}, {":scheme", "http"}, {":authority", "host"}};
-  EXPECT_TRUE(uhv->validateRequestHeaderMap(request_header_map));
+TEST_F(BaseHeaderValidatorTest, ValidateGenericHeaderKeyInvalidChar) {
+  HeaderString invalid_eascii{"x-foo\x80"};
+  auto uhv = createBase(reject_headers_with_underscores_config);
+
+  EXPECT_REJECT_WITH_DETAILS(uhv->validateGenericHeaderName(invalid_eascii),
+                             UhvResponseCodeDetail::get().InvalidCharacters);
 }
 
-TEST_F(HeaderValidatorTest, Http1ResponseHeaderMapValidation) {
-  auto uhv = create(empty_config, ::Envoy::Http::Protocol::Http11);
-  ::Envoy::Http::TestResponseHeaderMapImpl response_header_map{{":status", "200"}};
-  EXPECT_TRUE(uhv->validateResponseHeaderMap(response_header_map).ok());
+TEST_F(BaseHeaderValidatorTest, ValidateGenericHeaderKeyStrictValid) {
```

Comment:
What is the `Strict` in the test name? It looks like the same test as `ValidateGenericHeaderNameValid`

Reply:
This test sets the `restrict_http_methods` setting, which only allows RFC methods. I'm renaming this method now.